src/k8s/pkg/k8sd/api/cluster_join.go

```diff
@@ -22,6 +22,11 @@ func (e *Endpoints) postClusterJoin(s *state.State, r *http.Request) response.Re
 		return response.BadRequest(fmt.Errorf("invalid hostname %q: %w", req.Name, err))
 	}
 
+	context := r.Context()
+	if _, err := e.provider.MicroCluster().Status(context); err == nil {
+		return NodeInUse(fmt.Errorf("node %q is part of the cluster", hostname))
+	}
+
```

Comment:
This matches more closely with what the CLI does
https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8s/client/cluster.go#L13-L16

Reply:
@bschimke95 or @neoaggelos would you be able to confirm this is more appropriate than my first commit?

Reply:
That looks way more concise. Sorry, should have spotted this in the first review but I was tired :)